app/src/test/kotlin/com/example/cleanarchitecturesolution/rules/MainDispatcherRule.kt

```diff
@@ -0,0 +1,26 @@
+package com.example.cleanarchitecturesolution.rules
+
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.jupiter.api.extension.AfterEachCallback
+import org.junit.jupiter.api.extension.BeforeEachCallback
+import org.junit.jupiter.api.extension.ExtensionContext
+
+@DelicateCoroutinesApi
+@ExperimentalCoroutinesApi
+class MainDispatcherRule : AfterEachCallback, BeforeEachCallback {
```

Comment:
Wydaje mi się, że akurat w kontekście testowania domeny ten `Rule` nie jest wymagany. Może się natomiast przydać, gdy przejdziemy do testów View Modelu :)

Reply:
Tak - racja, ale chciałem skorzystać z kotlinx-corutines-test i bez tego Rule byłoby sporo duplikacji kodu, a tak to udało się zamknąć wszystko e jednym miejscu 🙂